mountpoint-s3/src/prefetch/backpressure_controller.rs

```diff
@@ -93,21 +89,22 @@ impl BackpressureController {
     /// will ensure that the read window size is enough to read this offset and that it is always close to `preferred_read_window_size`.
     pub async fn send_feedback<E>(&mut self, event: BackpressureFeedbackEvent) -> Result<(), PrefetchReadError<E>> {
         match event {
-            BackpressureFeedbackEvent::DataRead(length) => {
-                self.next_read_offset += length as u64;
+            // Note, that this may come from a backwards seek, so offsets observed by this method are not necessarily ascending
+            BackpressureFeedbackEvent::DataRead(offset, length) => {
+                let next_read_offset = offset + length as u64;
+                let remaining_window = self.read_window_end_offset.saturating_sub(next_read_offset);
```

Comment:
tiny nit: cast this to `usize` (and avoid `as u64` on the next line)

Reply:
this will be an incorrect cast on 32-bit, so I'd leave it as is, if you don't mind

(not super important, as I don't think we have tests for 32-bit or if anyone using that)

Reply:
From my perspective -- we shouldn't be using `usize`. It's safe if we always remember that length-s are `usize`, but event in `fuse.rs` interface [it's not](https://github.com/awslabs/mountpoint-s3/blob/main/mountpoint-s3/src/fuse.rs#L138). 

If you insist I can add a commit with a cast to `usize` though.

Reply:
I made it uniform, I guess it a drop in the ocean to use u64 here only. And may be you're right that this is wasteful, idk.